Harmony/Classes/Future/Observable+Time.swift

```diff
@@ -25,9 +25,9 @@ public extension Observable {
     ///   - queue: The queue to schedule the delay (by default the Main Queue).
     /// - Returns: A chained observable.
     func withDelay(_ interval: TimeInterval, queue: DispatchQueue = DispatchQueue.main) -> Observable<T> {
-        return Observable() { resolver in
-            queue.asyncAfter(deadline: .now() + interval) {
-                self.resolve(success: {value in
+        return Observable(parent: self) { resolver in
+            queue.asyncAfter(deadline: .now() + interval) { [weak self] in
+                self?.resolve(success: {value in
```

Comment:
Which is the Thread where the Future is resolved? 

- The one where the Future was executed first time?
- Inside the new Dispatch Queue?

Reply:
`then`/`fail` closures of `Future` and `Observable`, as all other functional methods, by default are called in the same thread as the `set` of error / value happens.

Only if the programmer defines a specific queue/executor as parameter of the used method, the  used thread is then switched.